Move to using ubuntu for docker

jcscottiii opened the pull request:
WIP

Commit:
Move to using ubuntu

mbland commented:
Hey @jcscottiii, as I pinged you in 18F/jekyll_pages_api_search#12, the "invalid character encoding" issue we were seeing in Docker should be resolved via #25.

Commit:
Add hash file for docker startup

Store the hash in order to make sure we adjust the time
appropriately.
On different hashes, assume the worst case and delete the
dependencies and start fresh. Set TRIES to high.
On a missing .bundle folder, set the TRIES high too.
Otherwise, don't have to try for so long.

jcscottiii commented:
@mbland it's ready now!

Commit:
WIP Trying to figure out Node.js install via nvm

Commit:
Fix nvm for dockerfile

NVM, Node, nor npm were available in the container.
It needed to be added to the bashrc

mbland commented:
Hey @jcscottiii, this is so cool! It totally works...except the smoke test fails during `./go ci_build`. Still, most everything else is pretty tight, so I'll merge this and file an issue for fixing that issue. Thanks so much for sticking with it!
